### PR TITLE
build(npm): cleaned up dependencies for better builds and npm7

### PR DESCRIPTION
**Pull request by 0-vortex**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [x] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [x] 🚩 Other

## Description

Quality of life improvements:
- added premanent lockfile for specified `package.engines`
- updated all deps that weren't breaking changes or unreleased betas
- separated concerns between dev and prod dependencies
- updated React to v17
- removed some unused packages that were likely artifacts or experiments

When switching between `npm` versions or if you broke your cache from stackoverflow you might need to do a `npm cache clean --force` to ensure the build is reproducible.

Another short term change some people might not agree with is forcing local `npm` to be verbose - this way we can properly record every issue but people not used to this might find it difficult to actually log. I've considered this is a lot more to gain in our current circumstances but open to tone it town or even remove it entirely if once it's not needed anymore. 

Given the complexity of the dependency tree it should help everyone, given some time to get comfortable with the logs, a huge insight on `npm` issues in general and distinguish steps needed to unblock people.

## Related Tickets & Documents

- #769 
- #883 
- #950 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

<details>
  <summary>Node v14</summary>

[asciicast](https://asciinema.org/a/VNuYO536e23IOhzi6NuRySYRs)

</details>

## Added tests?

- [x] 👍 yes, manual
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -5,13 +5,13 @@ import Config from "./config";
...
 import registerServiceWorker from "./registerServiceWorker";
-import OneGraphApolloClient from "onegraph-apollo-client";
+import ApolloClient from "apollo-boost";
 import {ApolloProvider} from "react-apollo";
 import api from "./lib/apiGraphQL";
 import {getAppVersion} from "./lib/appVersion";
 import {validateToken} from "./lib/validateToken";
 
-const apolloClient = new OneGraphApolloClient({
+const apolloClient = new ApolloClient({
```
Comment (bdougie) [inline src/index.js:14]: @sgrove do you have thoughts here. In order to leverage React v0.17 we are going with ApolloCliene over the OneGraphApolloClient.

Comment (dwwoelfel) [reply src/index.js:14]: What is it about `OneGraphApolloClient` that doesn't work with React v0.17? We can fix that in the onegraph-apollo-client package.

It's just a thin wrapper over `@apollo/client` that sets up auth and sets the correct path. It also makes it easier to set up subscriptions, but I'm not sure if you're using that.

If you do switch to plain ApolloClient, you'll want to make sure to set up the url and the auth, similar to what we do here: https://github.com/OneGraph/onegraph-client/blob/master/packages/onegraph-apollo-client/src/OneGraphApolloClient.js#L25

Comment (0-vortex) [reply src/index.js:14]: had BC updating to react17, didn't want to switch dependency resolution for reasons unrelated, so just followed documentation here: https://www.npmjs.com/package/onegraph-apollo-client which led me here: https://www.onegraph.com/docs/front_end_apollo.html - I'm fine with keeping what is but the docs definitely confused me, not being focused on onegraph/apollo other than having the frontend working. still confused on what should be updated or reverted :D

**Comment by 0-vortex** [comment]
@bdougie think netlify needs `npm@7` to proprely digest the new lockfile. 

Any chance I can play around with the build ?

**Pull request merged**